components/motor/gpio/motor_encoder.go

```diff
@@ -320,30 +210,26 @@ func (m *EncodedMotor) rpmMonitor() {
 	m.startedRPMMonitor = true
 	m.startedRPMMonitorMu.Unlock()
 
-	lastPosFl, _, err := m.encoder.Position(m.cancelCtx, encoder.PositionTypeUnspecified, nil)
+	lastPos, _, err := m.encoder.Position(m.cancelCtx, encoder.PositionTypeUnspecified, nil)
 	if err != nil {
 		panic(err)
 	}
-	lastPos := int64(lastPosFl)
 	lastTime := time.Now().UnixNano()
 
-	rpmSleep, rpmDebug := getRPMSleepDebug()
-	inRamp := false
-
 	for {
-		myRpmSleep := rpmSleep
-		if inRamp {
-			// if we're ramping up or down, make the loop faster
-			myRpmSleep /= 4
-		}
-		timer := time.NewTimer(myRpmSleep)
+		timer := time.NewTimer(50 * time.Millisecond)
 		select {
 		case <-m.cancelCtx.Done():
 			timer.Stop()
 			return
 		case <-timer.C:
 		}
-
+		m.stateMu.Lock()
+		if !m.state.regulated {
+			m.stateMu.Unlock()
+			continue
+		}
+		m.stateMu.Unlock()
 		pos, _, err := m.encoder.Position(m.cancelCtx, encoder.PositionTypeUnspecified, nil)
```

Comment:
It might be good to change these in this pr or the next call motor.Position and only have that function access the encoder with the specified type. Let potential bugs.